src/Models/CartRequirement.php

```diff
@@ -95,6 +95,68 @@ public function isEligible(): bool
         return $firstRequirementsMet && $secondRequirementsMet;
     }
 
+    /**
+     * Quantity for White Glove service product
+     * should match the quantity of the required product.
+     *
+     * @param CartItem $cartItem
+     * @return array
+     */
+    public function quantityIsEligible(CartItem $cartItem): array
+    {
+
+        if (empty($this->first_set_of_required_shopify_product_ids)) {
+            return $this->getStatusAndErrorMessage(null, null, true);
+        }
+
+        $nonWhiteGloveProductVariantsInCart = resolve(GoLoadUp::class)->getNonWhiteGloveProductVariantsInCart();
+
+        $firstRequirementsQuantityError = $this->catchQuantityMismatch($cartItem, $nonWhiteGloveProductVariantsInCart, $this->first_set_of_required_shopify_product_ids, );
+
+        if (! is_null($firstRequirementsQuantityError)) {
+            return $firstRequirementsQuantityError;
+        }
+
+        $secondRequirementsQuantityError = $this->catchQuantityMismatch($cartItem, $nonWhiteGloveProductVariantsInCart, $this->second_set_of_required_shopify_product_ids, );
+
+        if (! is_null($secondRequirementsQuantityError)) {
+            return $secondRequirementsQuantityError;
+        }
+
+        return $this->getStatusAndErrorMessage(null, null, true);
+
+    }
+
+    public function catchQuantityMismatch($cartItem, $nonWhiteGloveProductVariantsInCart, $set_of_required_shopify_product_ids): array|null
+    {
+        $requirementsMet = $nonWhiteGloveProductVariantsInCart
+            ->map(fn (CartItem $item) => $item->getProduct()?->id ?? null)
+            ->values()
+            ->intersect($set_of_required_shopify_product_ids);
+
+        foreach ($requirementsMet as $id) {
+
+            $retrievedCartItem = $nonWhiteGloveProductVariantsInCart->reject(function (CartItem $item) use ($id) {
+                return $item->getProduct()?->id != $id;
+            });
```

Comment:
I have never noticed the reject method until just now. So the idea behind rejection is to do the reverse of a filter? Meaning remove the products that are not allowed for white glove and return the remaining products?

Reply:
Yes, it removes any item for which the condition is true.